recipes-connectivity/iot-companion/iot-companion/iot-companion.sh

```diff
@@ -0,0 +1,652 @@
+			JqDump=$(jq -r 'to_entries[] | "\(.key)=\(.value)"' $AwsClientConfig > $JqOutput)
+			## Ensuring if jq is successful
+			[[ -s $JqOutput ]] && echo "$JqOutput exists and not empty" || echo "$JqOutput doesn't exist or is empty"
+		fi
+	fi
+else
+	echo "No stored hash for '$AwsClientConfig', creating hash file"
+	sha256sum "$AwsClientConfig" > "$AwsClientConfig.sha256"
+
+	echo "Creating jq dump file for first time"
+	# Creating jq dump file for first time
+	JqDump=$(jq -r 'to_entries[] | "\(.key)=\(.value)"' $AwsClientConfig > $JqOutput)
+fi
+
+## Ideas for additional sectionso
+
+# General 
+# versions      show Versions for all clients
+
+# aws shadow            show Current Shadow
+# aws onbaording        show Onboarding Status
+# aws log show 	        show the log database
+# aws log               use the loggig to cloud feature
+# aws license           show license manager feature and configuration
+
+# net                   network and connection related topics (selfcheck, IP, dns, ping)
+
+# wan                   wan related informations and setting (ppp, ethernet, wifi)
+
+# New Section "hw"      show all HW related information (OEM, Board, Revision, Serial, MAC, ..)
+# New section "hsm"     show all security features and status of hsm/tpm/se
+
+# phytec		show phytec tools (board-info, ... )
+
+# developer 		show how to install developer tools
+
+# security		show security related infos (SSHD, Keys, hotp, )
+
+############################################################################################################
+#<<< CONFIGURATION PARSERS 
+
+# Source jq dump file to retrieve configuration parameters
+if . $JqOutput; then
+	THING_NAME=$thing_name
+	ENDPOINT=$endpoint
+
+	ALLOW_LIST_FILE_DOWNLOAD=$maintenance_task_download_whitelist_path
+	FILE_DOWNLOAD_TEMP_DIR=$maintenance_task_temp_download_dir
+
+	ALLOW_LIST_COMMAND_EXCUTION=$maintenance_task_command_whitelist_path
+
+	REMOTE_MANAGER_CONFIG_DIR=$remote_manager_config_dir
+	REMOTE_MANAGER_CONFIG_FILE=$remote_manager_config_file
+	REMOTE_MANAGER_CONFIG_FILE=$REMOTE_MANAGER_CONFIG_DIR""$REMOTE_MANAGER_CONFIG_FILE
+
+	RAUC_HAWKBIT_CONFIG_DIR=$rauc_hawkbit_client_config_dir
+	RAUC_HAWKBIT_CONFIG_FILE=$rauc_hawkbit_client_config_file
+	RAUC_HAWKBIT_CONFIG_FILE=$RAUC_HAWKBIT_CONFIG_DIR""$RAUC_HAWKBIT_CONFIG_FILE
+
+	RAUC_HAWKBIT_BUNDLE_DOWNLOAD_LOCATION=$(awk -F "=" '/bundle_download_location/ {print $2}' $RAUC_HAWKBIT_CONFIG_FILE)
+
+	SSH_PUB_KEY_DIR=$ssh_pub_key_dir
+	SSH_PUB_KEY_FILE=$ssh_pub_key_file
+	SSH_PUB_KEY_FILE=$SSH_PUB_KEY_DIR""$SSH_PUB_KEY_FILE
+else
+	echo "Couldn't find the jq dump file! Re-run the script to create new jq dump file. Please don't delete the jq_dump file manually."
+fi
+
+#<<< CONFIGURATION PARSERS
+
+#<<< usage 
+#--------------------------------------------------------------------------------------------
+script_dir=$(cd "$(dirname "${BASH_SOURCE[0]}")" &>/dev/null && pwd -P)
+usage_header(){
+
+cat<<EOF
```

Comment:
For printing output to the screen, please use `printf` instead of `echo` or `cat` in this case.